### PR TITLE
Replace map components

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

This PR replaces the prior map components to use Google Maps. On the main map, we use the JS API; for the details page we use the static map APIs, which renders an image with a marker.

It also adds a new implementation of the clustering using markerclusterer -- https://developers.google.com/maps/documentation/javascript/marker-clustering#adding-a-marker-clusterer -- and adjusts the new map to make use of the existing disambiguation popup created in #230 

Connects #264 

## Demo

Note that the facilities in the multiple facility popup are all at 0,0, a spot for which there is no satellite imagery from the Google Static Map API. I don't believe that would be the case for actual facility locations.

facilities
multiple

## Notes

I also removed the prior mapping libraries & components.

One notable change here is that instead of setting data from GeoJSON directly, we are creating markers for each facility in a loop and then placing them in the markerclusterer. This was done specifically for the markerclusterer, but I don't know whether it will be more or less performant than the prior implementation.

Another notable change is that it replaces the prior styling with the default Google Maps markers. I did not yet adjust the marker in the main map to indicate when it has been selected, but I will try to work on that in a follow up issue. (For now users can see which facility is selected by seeing it in the details map).

I opted to use https://github.com/google-map-react/google-map-react as the wrapper for this because it seemed to have the best combination of: still maintained, seems widely used, and has an unobtrusive API. It's MIT licensed per

- https://github.com/google-map-react/google-map-react/tree/1.1.2#license
and
- https://github.com/google-map-react/google-map-react/blob/1.1.2/package.json#L39

The other two libraries I looked at were

- https://github.com/tomchentw/react-google-maps
and
- https://github.com/JustFly1984/react-google-maps-api

react-google-maps has slightly more GitHub stars than google-map-react, but is no longer actively maintained and has a long backlog of issues and pull requests. It does have nice documentation but I was a little wary of using something that's abandoned.

react-google-maps-api is newer and is actively maintained, but it doesn't seem to be very heavily used so it made me worry that we'd encounter bugs that hadn't yet been discovered in the library itself.

A fourth possibility I considered was just using handling all the map instantiation in lifecycle methods and not using a library at all. However, there were enough nice React-y things from google-map-react that it seemed better to keep it. (For just one example: you can render arbitrary React components at specific lat lngs on the map by passing them as children to the Map component, then giving them lat/lng props).

## Testing

- get this branch, then `./scripts/update`
- visit the main map page on 6543 and use the map and verify that it works correctly, without any catastrophic bugs

(We can immediately fix minor bugs in follow up issues.)

### CODE REVIEW

File: src/django/oar/settings.py [hunk 1]
```diff
@@ -235,6 +235,8 @@
     raise ImproperlyConfigured(
         'Invalid GOOGLE_GEOCODING_API_KEY provided, must be set')
 
+GOOGLE_MAPS_KEY = os.getenv('GOOGLE_GEOCODING_API_KEY')
```
Comment (kellyi) [inline src/django/oar/settings.py]: @jwalgran here is how I was getting the api key.

**Comment by jwalgran** [comment]
We chatted outside this PR about there being another React wrapper for Google maps that you opted not to use. I'd like to reference that in the Notes section.

**Comment by kellyi** [comment]
> We chatted outside this PR about there being another React wrapper for Google maps that you opted not to use. I'd like to reference that in the Notes section.

Sounds good. Adding the comment here and then I'll paste it above:

I opted to use https://github.com/google-map-react/google-map-react as the wrapper for this because it seemed to have the best combination of: still maintained, seems widely used, and has an unobtrusive API.

The other two libraries I looked at were

- https://github.com/tomchentw/react-google-maps
and
- https://github.com/JustFly1984/react-google-maps-api

react-google-maps has slightly more GitHub stars than google-map-react, but is no longer actively maintained and has a long backlog of issues and pull requests. It does have nice documentation but I was a little wary of using something that's abandoned.

react-google-maps-api is newer and is actively maintained, but it doesn't seem to be very heavily used so it made me worry that we'd encounter bugs that hadn't yet been discovered in the library itself.

A fourth possibility I considered was just using handling all the map instantiation in lifecycle methods and not using a library at all. However, there were enough nice React-y things from google-map-react that it seemed better to keep it. (For just one example: you can render arbitrary React components at specific lat lngs on the map by passing them as children to the Map component, then giving them lat/lng props).

File: src/app/src/util/constants.facilitiesMap.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import env from './env';
+
+export const GOOGLE_MAPS_API_KEY = env('REACT_APP_GOOGLE_MAPS_KEY');
+
+export const initialCenter = Object.freeze({
+    lat: 34,
+    lng: 5,
+});
+
+export const initialZoom = 1.5;
```
[the remark this answers is not in the document]
Comment (kellyi) [reply src/app/src/util/constants.facilitiesMap.js]: Sounds good. This was a holdover from the previous implementation, and I changed it to 1 here 
8c7e00e

**Pull request merged**